src/services/notificationService.ts

```diff
@@ -10,10 +11,37 @@ import { logger } from '../utils/logger';
 import { EMAIL_STATUSES, Notification } from '../entities/notification';
 import { SEGMENT_METADATA_SCHEMA_VALIDATOR } from '../utils/validators/segmentAndMetadataValidators';
 import { validateWithJoiSchema } from '../validators/schemaValidators';
-import { SegmentAnalyticsSingleton } from './segment/segmentAnalyticsSingleton';
 import { SendNotificationRequest } from '../types/requestResponses';
 import { StandardError } from '../types/StandardError';
 
+enum EmailNotificationId {
+  donationReceived = 'donationReceived',
+}
+
+const activityCreator = (payload: any, emailNotificationId: EmailNotificationId) => {
```

Comment:
1- Please use switch-case instead of `if` condition
2- don't use `any` type, and define the type definition for `payload`

Reply:
Done!